### DISCUSSION TITLE
Inspecting outside a browser

### DISCUSSION THREAD

**Discussion by Bickio** [statelyai/xstate#3007] [Ideas]
My use case for xstate involves running an instance on a node.js server, to keep track of shared state

I was hoping that it would be possible to inspect the statechart despite it not running in the browser

Is this a feature which might be considered? I would be happy to contribute if someone can point me in the right direction

**Comment by davidkpiano** [comment]
This is possible using WebSockets. See here for an example: https://github.com/statelyai/xstate/blob/main/packages/xstate-inspect/examples/server.ts
